### PR TITLE
use GOPROXY to speed up dependency management

### PR DESCRIPTION
**Pull request by vrothberg**
Set GOPROXY=https://proxy.golang.org in the Makefile to speed up
fetching dependencies.  Note that the go modules of
golangci/golangci-lint has some invalid version specifiers, so we had to
add some replacements in our go.mod to avoid getting 410's from the
GOPROXY.

Fixes: #2756
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 GO ?= go
 
+export GOPROXY=https://proxy.golang.org
```
Comment (saschagrunert) [inline Makefile]: What happens if the proxy is unavailable? Can we set it to something like `https://proxy.golang.org,direct` to still allow direct connections as fallback?

Comment (vrothberg) [reply Makefile]: That's the default in go v1.13, so I don't think we need to worry about the proxy being unavailable. The change is effectively an early opt-in for go < v1.13.

Regarding the fall back, I don't know if that's valid syntax for GOPROXY. I've checked the docs but didn't see a reference for that. Do you know if that's possible?

Comment (vrothberg) [reply Makefile]: `invalid $GOPROXY setting: cannot have comma` ... it's really a mutually exclusive setting. If the proxy is unavailable, it will fail.

Comment (saschagrunert) [reply Makefile]: Okay, so the comma separated list seems a go1.13 feature too, LGTM :)

**Comment by vrothberg** [comment]
@saschagrunert, do you know if the e2e tests are generally broken? Otherwise, I'll have a look at the failures here.

**Comment by saschagrunert** [comment]
> @saschagrunert, do you know if the e2e tests are generally broken? Otherwise, I'll have a look at the failures here.

I think https://github.com/cri-o/cri-o/pull/2759 is the closest solution right now, but we see new failures which does not seem to be reproducible locally (investigating this now too).

File: Makefile (hunk 1 above)
Comment (saschagrunert) [inline Makefile]: Do we also want to set `GOSUMDB=sum.golang.org`?

Comment (vrothberg) [reply Makefile]: Sounds good to me. However, it'll only take effect on go >= 1.13.

File: go.mod [hunk 2]
```diff
@@ -86,3 +86,19 @@ replace (
 	k8s.io/metrics => k8s.io/kubernetes/staging/src/k8s.io/metrics v0.0.0-20190807140250-28e88e3f19c3
 	k8s.io/sample-apiserver => k8s.io/kubernetes/staging/src/k8s.io/sample-apiserver v0.0.0-20190807140250-28e88e3f19c3
 )
+
+replace github.com/go-critic/go-critic v0.0.0-20181204210945-1df300866540 => github.com/go-critic/go-critic v0.3.5-0.20190526074819-1df300866540
```
Comment (saschagrunert) [inline go.mod]: Sorry to bump you again, but can we check if this is still needed with [golangci-lint v1.18.0](https://github.com/golangci/golangci-lint/releases/tag/v1.18.0)?

Comment (vrothberg) [reply go.mod]: That works like a charm now. Thanks for checking!

**Comment by saschagrunert** [comment]
This does not look good...

/retest

**Pull request merged**